src/otaclient/ota_core.py

```diff
@@ -1244,6 +1258,14 @@ def __init__(
         self.started = True
         logger.info("otaclient started")
 
+        # NOTE: not doing fstrim at startup when running as dynamic otaclient
+        if not _env.is_dynamic_client_running() and cfg.FSTRIM_AT_OTACLIENT_STARTUP:
+            fstrim_at_thread(
+                Path(CANONICAL_ROOT),
+                waited=False,
+                timeout=cfg.FSTRIM_AT_OTACLIENT_STARTUP_TIMEOUT,
+            )
```

Comment:
Just curious, why we can't use fstrim at startup when running as dynamic otaclient?

Reply:
Ah, the intention is to do fstrim on every ECU boots up, not at every OTA(this is handled in another place), as currently dynamic otaclient launching only occurs at OTA.